File: packages/app/src/services/software/definitions/evernote.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'evernote';
	public name = 'Evernote';
	public category = [SoftwareCategory.Productivity];
	public downloadName = 'Evernote-latest.exe';
	public icon = 'evernote.png';
	public homepage = 'https://evernote.com';

	public async resolveDownloadUrl() {
		return ok('https://win.desktop.evernote.com/builds/Evernote-latest.exe');
	}
}

File: src/lib/software/evernote.py
from src.lib.software import BaseSoftware

class Evernote(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'Evernote-latest.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://win.desktop.evernote.com/builds/Evernote-latest.exe')
